lib/Runtime/Library/SubString.cpp

```diff
@@ -37,6 +37,13 @@ namespace Js
         const char16 * subString = string->GetString() + start;
         void const * originalFullStringReference = string->GetOriginalStringReference();
 
+#if SYSINFO_IMAGE_BASE_AVAILABLE
+        AssertMsg(AutoSystemInfo::IsJscriptModulePointer((void*)originalFullStringReference)
+            || recycler->IsValidObject((void*)originalFullStringReference)
+            || VirtualTableInfo<PropertyRecord>::HasVirtualTable((void*)originalFullStringReference) && ((PropertyRecord*)originalFullStringReference)->IsBound(),
```

Comment:
Please wrap the nested && clause in parens, otherwise it's potentially confusing (and clang will shout at you)